pyanaconda/modules/network/kickstart.py

```diff
@@ -31,6 +33,11 @@ def parse(self, args):
         if hostname_only_command:
             retval.bootProto = ""
 
+        if hasattr(retval, "hostname") and retval.hostname:
+            (result, reason) = is_valid_hostname(retval.hostname)
+            if not result:
+                raise KickstartError("Invalid hostname '{}'. {}".format(retval.hostname, reason))
```

Comment:
From kickstart.py of the Storage module:
```

raise KickstartParseError(_("File system type \"{}\" given in autopart command is "
                            "invalid.").format(self.fstype), lineno=self.lineno)

```

You should raise the `KickstartParseError` and specify the line number.